src/Intl/ListFormatter/IntlListFormatter.php

```diff
@@ -0,0 +1,206 @@
+ *
+ * @author Ayesh Karunaratne <ayesh@aye.sh>
+ *
+ * @internal
+ */
+class IntlListFormatter
+{
+    public const TYPE_AND = 0;
+    public const TYPE_OR = 1;
+    public const TYPE_UNITS = 2;
+
+    public const WIDTH_WIDE = 0;
+    public const WIDTH_SHORT = 1;
+    public const WIDTH_NARROW = 2;
+
+    /**
+     * @var string
+     */
+    private $locale;
+    /**
+     * @var int
+     */
+    private $type;
+    /**
+     * @var int
+     */
+    private $width;
+
+    protected static $listPatterns = [
+        'en' => [
+            'listPattern-type-standard' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, and {1}',
+                2 => '{0} and {1}',
+            ],
+            'listPattern-type-or' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, or {1}',
+                2 => '{0} or {1}',
+            ],
+            'listPattern-type-or-narrow' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, or {1}',
+                2 => '{0} or {1}',
+            ],
+            'listPattern-type-or-short' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, or {1}',
+                2 => '{0} or {1}',
+            ],
+            'listPattern-type-standard-narrow' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, {1}',
+                2 => '{0}, {1}',
+            ],
+            'listPattern-type-standard-short' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, & {1}',
+                2 => '{0} & {1}',
+            ],
+            'listPattern-type-unit' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, {1}',
+                2 => '{0}, {1}',
+            ],
+            'listPattern-type-unit-narrow' => [
+                'start' => '{0} {1}',
+                'middle' => '{0} {1}',
+                'end' => '{0} {1}',
+                2 => '{0} {1}',
+            ],
+            'listPattern-type-unit-short' => [
+                'start' => '{0}, {1}',
+                'middle' => '{0}, {1}',
+                'end' => '{0}, {1}',
+                2 => '{0}, {1}',
+            ],
+        ],
+    ];
+
+    public function __construct(
+        string $locale,
+        int $type = self::TYPE_AND,
+        int $width = self::WIDTH_WIDE
+    ) {
+        $exceptionClass = PHP_VERSION_ID >= 80000 ? \ValueError::class : \InvalidArgumentException::class;
+        if ($locale !== 'en' && strpos($locale, 'en') !== 0) {
+            throw new $exceptionClass('Invalid locale, only "en" and "en-*" locales are supported');
```

Comment:
I have some mixed feelings about this. Since it's using the en-US set, for en-gb it would return a different result than the real class.